benches/min_spanning_tree.rs

```diff
@@ -9,52 +9,100 @@ use test::Bencher;
+fn min_spanning_tree_kruskal_praust_dir_bench(bench: &mut Bencher) {
     let a = digraph().praust_a();
     let b = digraph().praust_b();
 
-    bench.iter(|| (min_spanning_tree(&a), min_spanning_tree(&b)));
+    bench.iter(|| (iterate_mst_kruskal(&a), iterate_mst_kruskal(&b)));
 }
 
 #[bench]
-fn min_spanning_tree_full_undir_bench(bench: &mut Bencher) {
+fn min_spanning_tree_kruskal_full_undir_bench(bench: &mut Bencher) {
     let a = ungraph().full_a();
     let b = ungraph().full_b();
 
-    bench.iter(|| (min_spanning_tree(&a), min_spanning_tree(&b)));
+    bench.iter(|| (iterate_mst_kruskal(&a), iterate_mst_kruskal(&b)));
 }
 
 #[bench]
-fn min_spanning_tree_full_dir_bench(bench: &mut Bencher) {
+fn min_spanning_tree_kruskal_full_dir_bench(bench: &mut Bencher) {
     let a = digraph().full_a();
     let b = digraph().full_b();
 
-    bench.iter(|| (min_spanning_tree(&a), min_spanning_tree(&b)));
+    bench.iter(|| (iterate_mst_kruskal(&a), iterate_mst_kruskal(&b)));
 }
 
 #[bench]
-fn min_spanning_tree_petersen_undir_bench(bench: &mut Bencher) {
+fn min_spanning_tree_kruskal_petersen_undir_bench(bench: &mut Bencher) {
     let a = ungraph().petersen_a();
     let b = ungraph().petersen_b();
 
-    bench.iter(|| (min_spanning_tree(&a), min_spanning_tree(&b)));
+    bench.iter(|| (iterate_mst_kruskal(&a), iterate_mst_kruskal(&b)));
 }
 
 #[bench]
-fn min_spanning_tree_petersen_dir_bench(bench: &mut Bencher) {
+fn min_spanning_tree_kruskal_petersen_dir_bench(bench: &mut Bencher) {
     let a = digraph().petersen_a();
     let b = digraph().petersen_b();
 
-    bench.iter(|| (min_spanning_tree(&a), min_spanning_tree(&b)));
+    bench.iter(|| (iterate_mst_kruskal(&a), iterate_mst_kruskal(&b)));
+}
+
+#[bench]
+fn min_spanning_tree_prim_praust_undir_bench(bench: &mut Bencher) {
+    let a = ungraph().praust_a();
+    let b = ungraph().praust_b();
+
+    bench.iter(|| (iterate_mst_prim(&a), iterate_mst_prim(&b)));
+}
+
+#[bench]
+fn min_spanning_tree_prim_full_undir_bench(bench: &mut Bencher) {
+    let a = ungraph().full_a();
+    let b = ungraph().full_b();
+
+    bench.iter(|| (iterate_mst_prim(&a), iterate_mst_prim(&b)));
+}
+
+#[bench]
+fn min_spanning_tree_prim_petersen_undir_bench(bench: &mut Bencher) {
+    let a: Graph<(), (), Undirected> = ungraph().petersen_a();
+    let b = ungraph().petersen_b();
+
+    bench.iter(|| (iterate_mst_prim(&a), iterate_mst_prim(&b)));
+}
+
+fn iterate_mst_kruskal<G>(g: G) -> bool
+where
+    G: Data + IntoEdges + IntoNodeReferences + IntoEdgeReferences + NodeIndexable,
+    G::NodeWeight: Clone,
+    G::EdgeWeight: Clone + PartialOrd,
+{
+    let mst = min_spanning_tree(g);
+    mst.into_iter().all(|_| true)
+}
+
+fn iterate_mst_prim<G>(g: G) -> bool
+where
+    G: Data + IntoEdges + IntoNodeReferences + IntoEdgeReferences + NodeIndexable,
+    G::NodeWeight: Clone,
+    G::EdgeWeight: Clone + PartialOrd,
+{
+    let mst = min_spanning_tree_prim(g);
+    mst.into_iter().all(|_| true)
```

Comment:
Good catch. Using `black_box` is the best bet to avoid it getting optimised away.

```suggestion
fn iterate_mst_kruskal<G>(g: G)
where
    G: Data + IntoEdges + IntoNodeReferences + IntoEdgeReferences + NodeIndexable,
    G::NodeWeight: Clone,
    G::EdgeWeight: Clone + PartialOrd,
{
    for e in min_spanning_tree(g) {
        std::hint::black_box(e);
    }
}

fn iterate_mst_prim<G>(g: G)
where
    G: Data + IntoEdges + IntoNodeReferences + IntoEdgeReferences + NodeIndexable,
    G::NodeWeight: Clone,
    G::EdgeWeight: Clone + PartialOrd,
{
    for e in min_spanning_tree_prim(g) {
        std::hint::black_box(e);
    }
```